programs/mpl-core/src/state/update_authority.rs

```diff
@@ -48,59 +40,6 @@ impl UpdateAuthority {
         CheckResult::CanApprove
     }
 
-    /// Validate the create lifecycle event.
-    pub(crate) fn validate_create(
```

Comment:
Note that I reused some of this logic in the add/remove from collection handling on the staging branch.  It would be nice if we could apply some of this simplification to that branch, but not sure how much can be reused because adding/removing from a collection does not constitute a lifecycle event.